lib/factory_factory_girl/generators/model_generator.rb

```diff
@@ -18,21 +18,19 @@ def factory_attributes
       end
 
       def set_column(attribute)
-        match_results = rules.map do |rule|
-          if attribute.name.match(rule[:rule])
-            rule
-          end
-        end.compact
+        applied_rule = get_match_results(rules, attribute).first
+        return default_value(attribute) unless applied_rule
+        return "{ #{applied_rule[:function]} }" unless applied_rule[:value]
```

Comment:
Thanks. 

I can revert the ```if``` back to the block.  I'll go ahead and close this pull request for now as it will be a few days before I get time to get back to this.

I'll plan on opening a new pull request on Tuesday or Wednesday.